### ISSUE TITLE
Missing gradients?

### ISSUE THREAD

**Issue by davidsonic** [TIGER-AI-Lab/ConsistI2V#26]
Hi, Thank you for the intersting work. I tried to print out the gradients regarding the model weights and found that there are certain layers whose gradients were None, as below. 

`INFO - __main__ - missing_gradients: ['down_blocks.0.conv3ds.0.time_emb_proj.weight', 'down_blocks.0.conv3ds.0.time_emb_proj.bias', 'down_blocks.0.conv3ds.1.time_emb_proj.weight', 'down_blocks.0.conv3ds.1.time_emb_proj.bias', 'down_blocks.0.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.0.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.0.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.0.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.1.conv3ds.0.time_emb_proj.weight', 'down_blocks.1.conv3ds.0.time_emb_proj.bias', 'down_blocks.1.conv3ds.1.time_emb_proj.weight', 'down_blocks.1.conv3ds.1.time_emb_proj.bias', 'down_blocks.1.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.1.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.1.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.1.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.2.conv3ds.0.time_emb_proj.weight', 'down_blocks.2.conv3ds.0.time_emb_proj.bias', 'down_blocks.2.conv3ds.1.time_emb_proj.weight', 'down_blocks.2.conv3ds.1.time_emb_proj.bias', 'down_blocks.2.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.2.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.2.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'down_blocks.2.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'down_blocks.3.conv3ds.0.time_emb_proj.weight', 'down_blocks.3.conv3ds.0.time_emb_proj.bias', 'down_blocks.3.conv3ds.1.time_emb_proj.weight', 'down_blocks.3.conv3ds.1.time_emb_proj.bias', 'up_blocks.0.conv3ds.0.time_emb_proj.weight', 'up_blocks.0.conv3ds.0.time_emb_proj.bias', 'up_blocks.0.conv3ds.1.time_emb_proj.weight', 'up_blocks.0.conv3ds.1.time_emb_proj.bias', 'up_blocks.0.conv3ds.2.time_emb_proj.weight', 'up_blocks.0.conv3ds.2.time_emb_proj.bias', 'up_blocks.1.conv3ds.0.time_emb_proj.weight', 'up_blocks.1.conv3ds.0.time_emb_proj.bias', 'up_blocks.1.conv3ds.1.time_emb_proj.weight', 'up_blocks.1.conv3ds.1.time_emb_proj.bias', 'up_blocks.1.conv3ds.2.time_emb_proj.weight', 'up_blocks.1.conv3ds.2.time_emb_proj.bias', 'up_blocks.1.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.1.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.1.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.1.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.1.tempo_attns.2.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.1.tempo_attns.2.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.conv3ds.0.time_emb_proj.weight', 'up_blocks.2.conv3ds.0.time_emb_proj.bias', 'up_blocks.2.conv3ds.1.time_emb_proj.weight', 'up_blocks.2.conv3ds.1.time_emb_proj.bias', 'up_blocks.2.conv3ds.2.time_emb_proj.weight', 'up_blocks.2.conv3ds.2.time_emb_proj.bias', 'up_blocks.2.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.tempo_attns.2.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.2.tempo_attns.2.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.conv3ds.0.time_emb_proj.weight', 'up_blocks.3.conv3ds.0.time_emb_proj.bias', 'up_blocks.3.conv3ds.1.time_emb_proj.weight', 'up_blocks.3.conv3ds.1.time_emb_proj.bias', 'up_blocks.3.conv3ds.2.time_emb_proj.weight', 'up_blocks.3.conv3ds.2.time_emb_proj.bias', 'up_blocks.3.tempo_attns.0.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.tempo_attns.0.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.tempo_attns.1.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.tempo_attns.1.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.tempo_attns.2.transformer_blocks.0.attn1.rotary_bias.relative_attention_bias.weight', 'up_blocks.3.tempo_attns.2.transformer_blocks.0.attn2.rotary_bias.relative_attention_bias.weight', 'mid_block.conv3ds.0.time_emb_proj.weight', 'mid_block.conv3ds.0.time_emb_proj.bias', 'mid_block.conv3ds.1.time_emb_proj.weight', 'mid_block.conv3ds.1.time_emb_proj.bias']`

I realize that these are all from temporal layers but not spatial layers, even though all the parameters are supposed to be trained. I've verified that requires_grad of all params have been set to True. Is this behavior expected?

The code to reproduce is attached below and inserted after backward loss in train.py. 
`missing_gradients = []
  for n, p in unet.named_parameters():
      if p.requires_grad and p.grad is None:
          missing_gradients.append(n)
  logger.info(f'missing_gradients: {missing_gradients}')
 `

**Comment by wren93** [comment]
Hi, I believe it's because those layers are not actually used in the model. See https://github.com/TIGER-AI-Lab/ConsistI2V/blob/main/consisti2v/models/videoldm_unet_blocks.py#L523, time embedding is not input into conv3d layers, so the time embedding projection layers are not trained. Same with the rotary bias layers: https://github.com/TIGER-AI-Lab/ConsistI2V/blob/main/consisti2v/models/videoldm_attention.py#L767

**Comment by davidsonic** [comment]
Thank you @wren93 for the pointers! I'll close this issue.
